app/controllers/projects/samples/metadata/fields_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Projects
+  module Samples
+    module Metadata
+      # Controller actions for Project Samples Metadata Fields Controller
+      class FieldsController < Projects::Samples::ApplicationController
+        respond_to :turbo_stream
+
+        # Validates metadata edit params and builds the expected update_service param prior to calling
+        # MetadataController and the metadata update_service
+
+        # Param is received as:
+        # params: {sample: {edit_field: {key: {old_key: new_key}, value: {old_value: new_value}}}
+        # Fields that have not been changed will have equal old and new
+        def update # rubocop:disable Metrics/AbcSize
+          authorize! @project, to: :update_sample?
+          metadata_update_params = ::Samples::Metadata::Fields::EditService.new(@project, @sample, current_user,
```

Comment:
**nitpick (blocking):** Can you update the `EditService` to actually edit the fields, so after validation etc it should call the `Samples::Metadata::UpdateService`